### PR TITLE
Adds config to add paths of jaxb.index 

### LINKED ISSUE #19861: JaxbFileRootBuildItem config
### Description

add possibility to add JaxbFileRootBuildItem from config.

### Implementation ideas

add a config
quarkus.jaxb.jaxb-index-path=/path1,/path2
config generate JaxbFileRootBuildItem in jaxbProcessor.
modify the test jaxb test to add an application.properties with the config to point to jaxb.index.

### PR DESCRIPTION
**Pull request by netodevel**
This PR adds a new property to extension jaxb.

Issue: #19861

quarkus.jaxb.jaxb-index-path=com/example,com/test

### CODE REVIEW

**Comment by netodevel** [comment]
@dufoli @geoand 
guys, can you help me with the code review of this pr?

File: extensions/jaxb/pom.xml [hunk 1]
```diff
@@ -9,6 +9,10 @@
         <relativePath>../pom.xml</relativePath>
     </parent>
     <modelVersion>4.0.0</modelVersion>
+    <properties>
+        <maven.compiler.target>11</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
+    </properties>
```
Comment (gastaldi) [inline extensions/jaxb/pom.xml:15]: Remove this snippet

**Review by gastaldi — COMMENTED** [review]
Can you elaborate on a use case where having an external `jaxb.index` would be used? 

AFAIK the JAXB extension already indexes JARs containing this file if they belong to the application

**Comment by gastaldi** [comment]
> Can you elaborate on a use case where having an external `jaxb.index` would be used?
> 
> AFAIK the JAXB extension already indexes JARs containing this file if they belong to the application

Maybe this is more a question to @dufoli

File: extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java [hunk 2]
```diff
@@ -275,6 +278,18 @@ void processAnnotationsAndIndexFiles(
...
+            List<JaxbFileRootBuildItem> fileRootBuildItems = jaxbConfig.indexPath.get().stream()
+                    .map(JaxbFileRootBuildItem::new).collect(Collectors.toList());
+
+            for (JaxbFileRootBuildItem i : fileRootBuildItems) {
+                try (Stream<Path> stream = iterateResources(applicationArchivesBuildItem, i.getFileRoot())) {
+                    stream.filter(p -> p.getFileName().toString().equals("jaxb.index"))
+                            .forEach(p1 -> handleJaxbFile(p1, resource, reflectiveClass));
+                }
+            }
+        }
```
Comment (gastaldi) [inline extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java:291]: This is exactly the same as the code below. Add to a separate List containing all the `JaxbFileRootBuildItem` elements before iterating in the code below?

File: extensions/jaxb/pom.xml (hunk 1 above)
Comment (netodevel) [reply extensions/jaxb/pom.xml]: Done.

File: extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java (hunk 2 above)
Comment (netodevel) [reply extensions/jaxb/deployment/src/main/java/io/quarkus/jaxb/deployment/JaxbProcessor.java]: Done.

**Comment by netodevel** [comment]
@gastaldi, @dufoli,

any feedback for this issue?
I see that there are errors in some tests, but I need to know if this issue goes forward, so I can spend time fixing the test.

**Comment by gastaldi** [comment]
@netodevel sorry, I'm on vacation until next week. I'd appreciate if @dufoli could answer https://github.com/quarkusio/quarkus/pull/20188#pullrequestreview-759249507 when possible

**Comment by netodevel** [comment]
@gastaldi, alright i wait, tks

**Review by gsmet — CHANGES_REQUESTED** [review]
I think it has value but it needs some tweaking.

But probably a good idea to get @dufoli to explain the usage it could have (might be useful to improve the javadoc of the config property).

**Comment by gsmet** [comment]
I'm going to close this one, not because of the quality of your work but because I have no idea of the usage of it so it's hard to make an informed decision on it.

It also needs a rebase and probably some additional tweaking. If someone finds this PR and has a need for it, please explain the usage and we will revive this PR.

Thanks for your efforts anyway!

**Pull request closed without merging**